hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -1012,6 +1021,18 @@ private void putHeader(byte[] dest, int offset, int onDiskSize,
       Bytes.putInt(dest, offset, onDiskDataSize);
     }
 
+    private void putHeader(ByteBuff buff, int onDiskSize,
```

Comment:
Seems we can abstract the two putHeader into a single one ?   the byte[] dest can also be regarded as a ByteBuff ?

Reply:
you mean wrap the dest like this?
    **ByteBuff.wrap(ByteBuffer.wrap(dest))**
Will create extra ByteBuffer and ByteBuff instance, not friendly to YGC?

Reply:
That's OK.